docker/o3de-dev-extras-dev/Dockerfile

```
@@ -84,20 +86,26 @@ RUN apt-get update && apt-get install -y \
     ros-humble-rviz2 \
     ros-humble-tf2-ros \
     ros-humble-urdfdom \
-    ros-humble-vision-msgs &&    rm -rf /var/lib/apt/lists/*
+    ros-humble-vision-msgs \
+    ros-humble-rmw-cyclonedds-cpp &&    rm -rf /var/lib/apt/lists/*
 
 ## Symlink clang version to non-versioned clang and set cc to clang
 RUN update-alternatives --install /usr/bin/cc cc /usr/bin/clang 100 && \
     update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang++ 100
 
+ADD https://api.github.com/repos/o3de/o3de/git/refs/heads/${O3DE_VERSION} o3de-version.json
+
 # Install o3de
 RUN git clone -b ${O3DE_VERSION} --single-branch https://github.com/o3de/o3de.git && \
     cd o3de && \
     git lfs install && \
-    git lfs pull && \
+    git lfs pull
+
+RUN cd o3de && \
```

I believe this extra layer isn't needed because get_python is directly tied to o3de hash 
```suggestion
    git lfs pull && \
```